### PR TITLE
feat: support new location for content config

### PR DESCRIPTION
**Pull request by ascorbic**
## Changes

Adds support for content config files in `src/content.config.*`. This means there no longer needs to be a `src/content` folder.

Fixes PLT-2649

## Testing

Config files moved for most fixtures, though with some legacy collections kept in the old location to ensure test coverage.

## Docs

Docs will need updating with the new location, as eventually will templates etc.

### CODE REVIEW

File: .changeset/thirty-clocks-jump.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'astro': patch
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply .changeset/thirty-clocks-jump.md:2]: This is in `next` though. Does that apply?

[the remark this answers is not in the document]
Comment (ematipico) [reply .changeset/thirty-clocks-jump.md:2]: It is. At the end `changeset` groups the changesets by entity (major, minor and patches), so regardless of where we are, we need to stick with semver so `changeset` can output a faithful changelog

File: packages/astro/src/content/utils.ts [hunk 2]
```diff
@@ -729,9 +729,16 @@ export function getContentPaths(
...
 }
 function search(fs: typeof fsMod, srcDir: URL) {
-	const paths = ['config.mjs', 'config.js', 'config.mts', 'config.ts'].map(
-		(p) => new URL(`./content/${p}`, srcDir),
-	);
+	const paths = [
+		'content.config.mjs',
+		'content.config.js',
+		'content.config.mts',
+		'content.config.ts',
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply packages/astro/src/content/utils.ts:736]: WHat is it about old collections that mean they do need this separate Vite server?

[the remark this answers is not in the document]
Comment (ematipico) [reply packages/astro/src/content/utils.ts]: It's still hard to grasp for me, @matthewp and @bholmesdev can definitely explain that better, but at the moment the content collections are loaded **before** the main vite server instance starts, which means that loading of the configuration files has all sort of gimmicks and hacks (see the observable pattern we use to subscribe to it).

This doesn't happen with the middleware and the actions, so I figured if we don't intend to support the old content collections in this new configuration file, we might be able to remove that awkward architecture and use a more mature and reliable one.

[the remark this answers is not in the document]
Comment (ascorbic) [reply packages/astro/src/content/utils.ts]: I've changed it to not support the legacy ones for now, and hopefully we can simplify things in future then

File: .changeset/thirty-clocks-jump.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+---
+
+Changes the default content config location from `src/content/config.*` to `src/content.config.*`.
+
+The previous location is still supported, but the new location is now preferred.
```
[the remark this answers is not in the document]
Comment (ematipico) [reply .changeset/thirty-clocks-jump.md:7]: @ascorbic should update the changeset, and mention that the new config **doesn't** old content collections? (At least, that's what I understood from one of your comments).

**Review by ematipico — APPROVED** [review]
Few things that I couldn't understand: 
- is `content/config.ts` considered deprecated?
- is there a plan for the "old" configuration file? If so, which one is it?

Code wise it looks good though

**Pull request merged**